base_layer/blockchain/src/blockchainstate.rs

```diff
@@ -22,45 +22,169 @@
-        }
+    // ToDo link to some config for the pruning horizon, its currently 5000.
+    pub fn new() -> Result<BlockchainState, StateError> {
+        fs::create_dir("./storage/state").unwrap();
+        let store = BlockchainState::build_db()?;
+        let mut headers = MerkleMountainRange::new();
+        headers.init_persistance_store(&"headers".to_string(), std::usize::MAX);
+        let mut outputs = MerkleMountainRange::new();
+        outputs.init_persistance_store(&"outputs".to_string(), 5000);
+        let mut range_proofs = MerkleMountainRange::new();
+        range_proofs.init_persistance_store(&"range_proofs".to_string(), 5000);
+        let mut kernels = MerkleMountainRange::new();
+        kernels.init_persistance_store(&"kernels".to_string(), std::usize::MAX);
+        let block_chain_state = BlockchainState {
+            headers,
+            outputs,
+            kernels,
+            range_proofs,
+            store,
+        };
+
+        Ok(block_chain_state)
+    }
+
+    fn build_db() -> Result<LMDBStore, DatastoreError> {
+        let builder = LMDBBuilder::new();
+        builder
+            .set_mapsize(5)
+            .set_path("./storage/state/")
+            //create for headers mmr
+            .add_database(&"headers_mmr_checkpoints".to_string())
+            .add_database(&"headers_mmr_objects".to_string())
+            .add_database(&"headers_init".to_string())
+            //create for outputs mmr
+            .add_database(&"outputs_mmr_checkpoints".to_string())
+            .add_database(&"outputs_mmr_objects".to_string())
+            .add_database(&"outputs_init".to_string())
+            //create for range_proofs mmr
+            .add_database(&"range_proofs_mmr_checkpoints".to_string())
+            .add_database(&"range_proofs_mmr_objects".to_string())
+            .add_database(&"range_proofs_init".to_string())
+            //create for kernels mmr
+            .add_database(&"kernels_mmr_checkpoints".to_string())
+            .add_database(&"kernels_mmr_objects".to_string())
+            .add_database(&"kernels_init".to_string())
+            .build()
+    }
+
+    /// Will update the pruning horizon to the new value for the outputs data store
+    pub fn change_pruning_horizon(&mut self, new_pruning_horizon: usize) {
+        self.outputs.set_pruning_horizon(new_pruning_horizon);
     }
 
     /// This function  will process a new block.
-    /// Note the block must have been validated by the chainstate before.
-    pub fn process_new_block(&self, new_block: &Block) -> Result<(), StateError> {
-        self.validate_new_block(new_block)
+    /// Note the block is consumed by the function.
+    pub fn process_new_block(&mut self, mut new_block: Block) -> Result<(), StateError> {
+        self.validate_new_block(&new_block)?;
+        self.prune_all_inputs(&new_block)?;
+        // All seems valid, lets add the objects to the state
+        let mut drainer = new_block.body.outputs.drain(..);
```

Comment:
Is there a reason you don't just 

`for output in new_block.body.outputs.drain(..) { ... } ` here?

Reply:
No specific reason but thats kinda temp code that will most likely change in the next pr